tests/xharness/SimpleTcpListener.cs

```diff
@@ -50,6 +67,58 @@ protected override void Start ()
 				}
 			}
 		}
+		
+		void StartTcpTunnel ()
+		{
+			if (!TunnelHoleThrough.Task.Result) { // do nothing until the tunnel is ready
+				throw new InvalidOperationException ("Tcp tunnel could not be initialized.");
+			}
+			bool processed;
+			try {
+				int timeout = 100;
+				var watch = new System.Diagnostics.Stopwatch ();
+				watch.Start (); 
+				while (true) {
+					try {
+						client = new TcpClient ("localhost", Port);
+						Log.WriteLine ("Test log server listening on: {0}:{1}", Address, Port);
+						// let the device know we are ready!
+						var stream = client.GetStream ();
+						var ping = Encoding.UTF8.GetBytes ("ping");
+						stream.Write (ping, 0, ping.Length);
+						break;
+
+					} catch (SocketException ex) {
+						if (timeout == 100 && watch.ElapsedMilliseconds > 20000) {
+							timeout = 250; // Switch to a 250ms timeout after 20 seconds
+						} else if (timeout == 250 && watch.ElapsedMilliseconds > 120000) {
+							// Give up after 2 minutes.
+							throw ex;
+						}
+						Thread.Sleep (timeout);
+					}
+				}
+				do {
+					client.ReceiveBufferSize = buffer.Length;
+					processed = Processing (client);
+				} while (!AutoExit || !processed);
+			} catch (Exception e) {
+				var se = e as SocketException;
+				if (se == null || se.SocketErrorCode != SocketError.Interrupted)
+					Console.WriteLine ("[{0}] : {1}", DateTime.Now, e);
+			} finally {
```

Comment:
Nit + question, we log anything but interrupted and ignore any other exception is this what you intended? 

```suggestion
			} catch (SocketException e) when (e.SocketErrorCode != SocketError.Interrupted) {
				Console.WriteLine ("[{0}] : {1}", DateTime.Now, e);
			} catch (Exception e) {
				// ??
			} finally {
```

Reply:
Yes, at it comes from the previous implementation.